### ISSUE TITLE
[CM12.1]-Iyokan severe lags on incoming calls

### ISSUE THREAD

**Issue by fred777** [LegacyXperia/local_manifests#1000]
This issue is similar to https://github.com/LegacyXperia/local_manifests/issues/933#issuecomment-147103799 but I'm unable to reopen it, so I started a new ticket.

On my Iyokan, there are severe lags almost always when using phone app. Especially on incoming calls it may take more than 6 seconds for the phone to be come responsive. Even via hardware key I am unable to answer the call, so the caller often hangs up in the mean time.

I'm on 12.1-20150825 (self-built) but this issue was present in earlier versions too.

Also for outgoing calls there frequently are long delays until the in-call-app is being shown+responsive.

I already checked my ART settings - small ART mode is disabled, all dex2oat filter settings are set to balanced - and wiped cache without success.

It does not seem to be a ZRAM or low mem issue (around 100MB free). I even tried disabling ZRAM -> didn't help.

In order to track this down, I installed a tool showing CPU+RAM+IO utilization as a screen overlay. (https://play.google.com/store/apps/details?id=ds.cpuoverlay)

What I find now is: each time just after resuming from standby, still on lock screen, there is heavy CPU (100% @ 1.2GHz) and IO (10..20MB/s) going on for around 10 seconds and system.ui is shown as being the most active app during that time. Free RAM is ok (around 100MB).

My guess now is, that there are some actions / services / intents triggered just after resume that slow everything down which affect incoming calls too, especially if the phone app is not cached but has to be restarted from internal flash.

Especially, I am wondering about SystemUI....

**Comment by realcatgirly** [comment]
Sometimes the systemui crashes and i have a black screen for 30 sec

**Comment by fred777** [comment]
The situation improved for me...

I disabled some GMS services, most noteworthy: those related to Wearable API because they consumed lots of CPU time just after resuming from standby.

CPU/IO spikes just after resume from standby seem to be gone now.

Also I did some build.prop tweaks, i.e. decreasing heapsize from 128m to 96m and increasing heaptargetutilization from 0.75 to 0.9 (according to https://01.org/android-ia/user-guides/android-memory-tuning-android-5.0-and-5.1)

ZRAM is still enabled and swappiness=10

I'm pretty confident now, but the issue still needs more long term observation before I am able to draw serious conclusions...

Regarding memory consumption, the shell command "dumpsys meminfo" came in handy. A.o., it tells whether incall app / dialer / phone is held in memory persistently.
